src/app/lightning/update-ln-payments.ts

```diff
@@ -0,0 +1,93 @@
+import { LndService } from "@services/lnd"
+import { baseLogger } from "@services/logger"
+import { LnPaymentsRepository } from "@services/mongoose/ln-payments"
+
+export const updateLnPayments = async (): Promise<true | ApplicationError> => {
+  let incompleteLnPayments = await LnPaymentsRepository().listIncomplete()
+  if (incompleteLnPayments instanceof Error) return incompleteLnPayments
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const lndListMethods = [lndService.listSettledPayments, lndService.listFailedPayments]
+  for (const listPaymentsFn of lndListMethods) {
```

Comment:
shouldn't this be a `for await`? otherwise the loop will wait for the internal await?

Reply:
Not sure, I assumed `for await ... of` only worked for async iterables and in this case I think `lndListMethods` is a sync iterable containing async items.

Maybe @dolcalmi / @samerbuna  / @daviroo knows as well. Would be great if we could get all the iterations to trigger simultaneously, but still resolve their internal `await fetchAndUpdatePayments` so we can check for errors.

Reply:
I was just looking some more into how this works ([this was helpful](https://stackoverflow.com/a/59695815)).

If we wanted to not wait for the internal `await` we could wrap the entire internal `for` loop logic in a promise and then fire these off using `Promise.all`. That would break the intent of this for loop though, since we are doing a synchronous update of the `incompletePayments` array to remove processed elements.

I'm not sure if there's a way to work on the `incompletePayments` object asynchronously and whether this optimization would be worth the added complexity at this point (premature optimization?)

Reply:
I think `for await (const res of items.map(e => somethingAsync(e))) ` method should work having `somethingAsync` as a function that wraps line 14 to 19. 

Also 
- `update-ln-payments.ts` is under lightning folder, `-ln-` is not necessary
- Please use `import { Lightning } from "@app"` and `Lightning.updatePayments` (to use wrapped functions in the cron)